tests/integration/test_automation/testParser.py

```diff
@@ -0,0 +1,507 @@
+import yaml
+from pathlib import Path
+from colorama import Fore, Back, Style
+
+class ACE:
+
+    # constructor function    
+    def __init__(self, time_stamp,address,  snoop, rwtype, srcline):
+        self.time_stamp = time_stamp
+        self.address = address
+        self.snoop = snoop
+        self.rwtype = rwtype
+        self.srcline = srcline
+
+class AXI:
+
+    # constructor function    
+    def __init__(self, time_stamp,address, rwtype,srcline):
+        self.time_stamp = time_stamp
+        self.address = address
+        self.rwtype = rwtype
+        self.srcline = srcline
+
+class SNOOP:
+
+    # constructor function    
+    def __init__(self, time_stamp,address, crresp, msgtype, srcline):
+        self.time_stamp = time_stamp
+        self.address = address
+        self.crresp = crresp
+        self.msgtype = msgtype
+        self.srcline = srcline
+
+class DATA_BUFFER_ACCESS:
+
+    def __init__(self, address, min_address, max_address):
+        self.address = address
+        self.min_address = min_address
+        self.max_address = max_address
+
+class ACE_LIST_CHECKER:
+
+    def __init__(self, theList,theDict, rw):
+        self.theList = theList
+        self.theDict = theDict
+        self.rw = rw
+ 
+
+    def checker (self, group_index, message_index, certainty):
+        message_found = False
+        snoop_search = self.theDict[test_yaml['test']['groups'][i]['messages'][j][self.rw]]
```

Comment:
where does test_yaml come from?
I suppose it is a global variable: why not adding a function which passes it to the class or passing it via the constructor?

Reply:
yeah, there are quite a few globals. I haven't focussed on removing globals on this  PR. I focussed on getting it all to work. But I can remove globals in this PR

Reply:
done

Reply:
OK